lib/widgets/autocomplete.dart

```diff
@@ -223,28 +230,57 @@ class ComposeAutocomplete extends AutocompleteField<ComposeAutocompleteQuery, Co
 }
 
 class _MentionAutocompleteItem extends StatelessWidget {
-  const _MentionAutocompleteItem({required this.option});
+  const _MentionAutocompleteItem({required this.option, required this.narrow});
 
   final MentionAutocompleteResult option;
+  final Narrow narrow;
 
   @override
   Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
     Widget avatar;
-    String label;
+    Widget label;
     switch (option) {
       case UserMentionAutocompleteResult(:var userId):
-        avatar = Avatar(userId: userId, size: 32, borderRadius: 3);
-        label = PerAccountStoreWidget.of(context).users[userId]!.fullName;
+        avatar = Avatar(userId: userId, size: 32, borderRadius: 3); // web uses 21px
+        label = Text(store.users[userId]!.fullName);
+      case WildcardMentionAutocompleteResult(:var wildcard):
+        avatar = const Icon(ZulipIcons.three_person, size: 29); // web uses 19px
+        label = wildcardLabel(wildcard, context: context, store: store);
     }
 
     return Padding(
       padding: const EdgeInsets.symmetric(horizontal: 16.0, vertical: 8.0),
       child: Row(children: [
         avatar,
         const SizedBox(width: 8),
-        Text(label),
+        label,
       ]));
   }
+
+  Widget wildcardLabel(WildcardMentionOption wildcard, {
+    required BuildContext context,
+    required PerAccountStore store,
+  }) {
+    final isDmNarrow = narrow is DmNarrow;
+    final isChannelWildcardAvailable = store.account.zulipFeatureLevel >= 247; // TODO(server-9)
+    final localizations = ZulipLocalizations.of(context);
+    final description = switch (wildcard) {
+      WildcardMentionOption.all || WildcardMentionOption.everyone => isDmNarrow
+        ? localizations.wildcardMentionAllDmDescription
+        : isChannelWildcardAvailable
+            ? localizations.wildcardMentionChannelDescription
+            : localizations.wildcardMentionStreamDescription,
+      WildcardMentionOption.channel => localizations.wildcardMentionChannelDescription,
+      WildcardMentionOption.stream => isChannelWildcardAvailable
+        ? localizations.wildcardMentionChannelDescription
+        : localizations.wildcardMentionStreamDescription,
+      WildcardMentionOption.topic => localizations.wildcardMentionTopicDescription,
+    };
+    return Text.rich(TextSpan(text: '${wildcard.canonicalString} ', children: [
+      TextSpan(text: description, style: TextStyle(fontSize: 12,
+        color: Colors.black.withValues(alpha: 0.8)))]));
```

Comment:
I assume we won't want to use a space character to separate the wildcard text and its description:

<img width="455" alt="image" src="https://github.com/user-attachments/assets/617fbb6c-d5a1-473a-92d4-8dfdb802effe" />

but that change can wait until a later pass where we align this with the Figma (pending #995), like https://github.com/zulip/zulip-flutter/pull/889/files#r1896061825